Implement capture for generators

NickeZ opened the pull request:
Not sure if you want this or not. I need it for a project I'm trying to convert to meson..

jpakkane commented on mesonbuild/backend/ninjabackend.py:
```diff
@@ -1813,6 +1813,20 @@ def generate_genlist_for_target(self, genlist, target, outfile):
+                    self.environment.get_build_dir(),
+                    capture=outfiles[0]
+                )
+                print(self.environment.get_build_dir())
```
Forgotten debug print?

jpakkane commented:
This is nice, it should get a mention in the release notes as well. Please add a snippet file for this.

NickeZ commented:
Removed the debug print, added snippet in release notes.

Please review this carefully. I basically copied code from `custom_target` that already had `capture`. It works for me at least..

nirbheek commented:
CI is failing because you didn't implement it for the VS backend.

NickeZ commented:
Hopefully fixed it. tried it on vs2015.

NickeZ commented:
Rebased, should be ready for code review/merge
